src/java.logging/share/classes/java/util/logging/Handler.java

```diff
@@ -337,7 +340,8 @@ public Level getLevel() {
      * @return true if the {@code LogRecord} would be logged.
      *
      */
-    public boolean isLoggable(LogRecord record) {
+    // JDK11 has a bug which makes null records cause NPE, despite the spec. Later versions fix it.
+    public boolean isLoggable(@Nullable LogRecord record) {
```

Comment:
So here it probably depends on what version of the JDK you want to support for jspecify/jdk.
The method specification mentions `null` for `record`, but the `@param` omits mentioning nullness.
The code clearly throws an NPE when `record` is null.
That issue was [fixed in JDK 13 it seems](https://github.com/openjdk/jdk/commit/e58a4da301c6a7216fbefff7836f9837019e26dc).
The [JDK 17 version](https://github.com/eisop/jdk/blob/6cb2a2123fe5564d5094bf201126fa6c26df6392/src/java.logging/share/classes/java/util/logging/Handler.java#L337) of the JDK we use in EISOP contains that fix. However, that specification is also used for JDK <13, so making it `@Nullable` is a bit dangerous.
Would it be bothersome to keep this `@NonNull`? It seems moot to ask whether `null` is loggable, as the answer is always `false`.

Reply:
It's definitely a mess. As best I could tell from a quick look, it's possible that no one ever passes `null` to `isLoggable` or, for that matter, to `publish`. So this may be mostly academic.

Looking into it a bit more, I see that many `publish` implementations begin with a call to `isLoggable`. So it's convenient for those two methods to have annotations that are _consistent_ so that the one can delegate to the other. Then again, with a tool like the Checker Framework (and probably with various other tools, with Kotlin as the notable exception), you would always be able to widen the parameter type of `isLoggable` when you override it. Moreover, if we think it's realistic that someone would pass `null` to `publish`, then we don't _want_ it to be "convenient" to pass that along to `isLoggable` if you're inherited an `isLoggable` implementation that is going to throw :)

I think I'll stick with what the doc says here, but I will probably start the eisop PR by declaring it as non-nullable :)